pkg/resource/deploy/step_generator.go

```diff
@@ -621,7 +621,7 @@ func (sg *stepGenerator) issueCheckErrors(new *resource.State, urn resource.URN,
 }
 
 func (sg *stepGenerator) getResourceProvider(
```

Comment:
I get that these are all judgement calls but I kind of think that `getResourceProvider` should still return an error and its callers should turn them into results after logging errors. In my mind things like `getResourceProvider` imply some level of purity - I would be a little surprised if `getResourceProvider` logged an error, since I'd expect it to 1) be idempotent and 2) do little more than look up some state in a bag somewhere, which is basically what it's doing.

I think we should try to preserve the purity of functions like this one instead of resultifying it outright; that way we can read "this function results `result.Result`; therefore, it must have the effect of logging messages. I should not call this more than once."

Reply:
Discussed offline.  Will rename to loadResourceProvider to better indicate this is doing something heavy that warrants reporting an issue to the user.  We also discussed that it really should be teh case that if something is `Result`-returning, that strongly indicates that they took into their own hands makign sure the user was informed of an issue.